cert-tools/snapstore/src/snapstore/info.py

```diff
@@ -0,0 +1,178 @@
+"""
+Retrieve snap info through the snap Store API
+
+Ref:
+- https://api.snapcraft.io/docs/refresh.html
+- https://api.snapcraft.io/docs/info.html
+"""
+
+from argparse import ArgumentParser
+import json
+from typing import Iterable, Optional
+
+from snapstore.client import SnapstoreClient
+
+
+class Info:
+
+    def __init__(self, client: SnapstoreClient):
+        self.client = client
+
+    def raw_info_snap(
+        self,
+        snap: str,
+        architecture: Optional[str] = None,
+        store: Optional[str] = None,
+        fields: Optional[Iterable[str]] = None
+    ) -> dict:
+        """
+        Return info for a specific snap, as retrieved from the
+        `v2/snaps/info/{snap}` endpoint of the snap Store API.
+
+        The result contains a `channel-map` with information about
+        the different channels on which the snap is published.
+
+        Ref: https://api.snapcraft.io/docs/info.html
+        """
+        params = {}
+        if architecture:
+            params["architecture"] = architecture
+        if fields:
+            params["fields"] = ",".join(field.strip() for field in fields)
+        return self.client.get(
+            endpoint=f"v2/snaps/info/{snap}",
+            params=params,
+            store=store,
+        )
+
+    def raw_refresh_one(
+        self,
+        snap: str,
+        channel: str,
+        architecture: str,
+        store: Optional[str] = None,
```

Comment:
I've noticed that when there is a dedicated snap store, and the checkbox frontend snap is not included in the inclusion list, the API will fail. At least, `info` does.

We could maybe retry w/o `store` in case of failure? 

If we go that way though, the following `snap download` should have the `SNAP_STORE` variable unset... so it's a bit trickier than a simple retry.

Reply:
We agreed to follow up on this.